### PR TITLE
Changes in grid_search() for correct get_best_score()

### PR DESCRIPTION
**Pull request by rednevaler**
Before submitting a pull request, please do the following steps:

1. Read [instructions for contributors](https://catboost.ai/docs/concepts/development-and-contributions.html).
2. Run `ya make` in catboost folder to make sure the code builds.
3. Add tests that test your change.
4. Run tests using `ya make -t -A` command.
5. If you haven't already, complete the [CLA](https://yandex.ru/legal/cla/).

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=ru

### CODE REVIEW

File: catboost/python-package/catboost/_catboost.pyx [hunk 1]
```diff
@@ -4458,6 +4476,19 @@ cdef class _CatBoost:
                 cv_results
             )
             result_metrics.add(name)
+            metric_name = results.CvResult[metric_idx].Metric
+            if choose_by_train_test_split == False:
```
Comment (nikitxskv) [inline catboost/python-package/catboost/_catboost.pyx]: `if not choose_by_train_test_split:`

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1165958](https://a.yandex-team.ru/review/1165958)

File: catboost/python-package/ut/medium/test.py [hunk 2]
```diff
@@ -2504,6 +2504,52 @@ def test_grid_search_aliases(task_type):
...
+                    "eval_metric": "AUC",
+                    "task_type": task_type,
+                    "custom_metric": ["CrossEntropy", "F1"]
+                }
+            )
+            feature_border_type_list = ['Median', 'Uniform', 'UniformAndQuantiles', 'MaxLogSum']
+            one_hot_max_size_list = [4, 7, 10]
+            iterations_list = [5, 7, 10]
+            border_count_list = [4, 10, 50, 100]
+            results = model.grid_search(
```
Comment (nikitxskv) [inline catboost/python-package/ut/medium/test.py]: `results = model.grid_search(` ->`model.grid_search(`

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
Check [1165958](https://a.yandex-team.ru/review/1165958) failed: 11 test(s) broken

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1166120](https://a.yandex-team.ru/review/1166120)

**Comment by arcadia-devtools** [comment]
Check [1166120](https://a.yandex-team.ru/review/1166120) passed

**Pull request merged**